classes/AddressFormat.php

```diff
@@ -438,6 +438,11 @@ public static function generateAddress(Address $address, $patternRules = [], $ne
         $addressFields = AddressFormat::getOrderedAddressFields($address->id_country);
         $addressFormatedValues = AddressFormat::getFormattedAddressFieldsValues($address, $addressFields);
 
+        Hook::exec('addressFormatGenerateAddressBefore', [
+            'controller' => Context::getContext()->controller,
+            'patternRules' => &$patternRules
+        ]);
```

Comment:
```suggestion
        Hook::exec('addressFormatGenerateAddressBefore', [
            'address' => $address,
            'patternRules' => &$patternRules
        ]);
```

Reply:
I think we better not remove the controller, because it lets us to check, where the hook is executed.
Also I think we better add address fileds whichh are country specific and address formated values which are address specific. All these together gives developers a good controll on what is presented in front end apps and how is it presented.

Reply:
> I think we better not remove the controller, because it lets us to check, where the hook is executed.

You can access the current controller using the `Context`